### DISCUSSION TITLE
will reading tx ty of sprite's worldTransform consume more time?

### DISCUSSION THREAD

**Discussion by lancety** [pixijs/pixijs#7506] [General]
hi guys, I have a culling method to check if children sprite is out of screen, if yes then set renderable = false otherwise = true. This change improved the fps from about 9ms to 6-7ms.
It is expected that "updateTransform" will take same time compare to the code without this culling change. And the render time for these sprite is much smaller because 1/5 - 1/3 sprites are out of screen which wont be rendered now. BUT the culling logic takes much longer than expected. 

There are normally 5-7 containers, each one have 300 - 500 sprites. The culling loop should totally process around 3000 sprites by comparing the tx ty of worldTransform with the screen width and height (viewState.camScope_meter).
The green block in the debugger profile is the culling method, which is more than 0.46, sometime almost 1ms.  To me this kind of calculation should be lightening fast, by replacing the tx, ty of worldTransform with fix int, the function runs very quick and the profiler even wont show it because it is too quick. And no matter where I put the tx ty read from worldTransform, the time taken nearly 100ms always shows on the line of "const {tx, ty} = worldTransform", is it really the worldTransform get property logic taking that much of time?

![image](https://user-images.githubusercontent.com/5547558/119326431-393fe500-bcc5-11eb-9d40-707a17ecf6f4.png)
![image](https://user-images.githubusercontent.com/5547558/119326498-4b218800-bcc5-11eb-892e-5b9fa647b1d6.png)

---------------------------- question 2 related to this topic

I am thinking about one question want to confirm which is related to this transform stuff as well @ivanpopelyshev ,  In my game, in 4th level children containers, there are 3000 sprites & graphics updateTransform takes 1.25  - 1.75ms,  the https://pixijs.io/bunny-mark/ "updateTransform of 3000 bunnies is about 0.33ms.    is it affected by:
- deeper nested children containers/sprites?
- scale setting of the parent containers?
- does graphics takes more time in updateTransform logic? there are few graphics (less than 10)

![image](https://user-images.githubusercontent.com/5547558/119329012-18c55a00-bcc8-11eb-8216-3fef02d0e139.png)
![image](https://user-images.githubusercontent.com/5547558/119329037-224ec200-bcc8-11eb-8eeb-3adb8c93abff.png)

---------------------------- question 3 related to this topic

Since this culling is trying to improve peroformance, I am also wondering if I should use pixi-layer, my understanding is if these  sprites in each region(container) all added to one layer, the sprite batch will run less loop in each ticker render loop, is it correct? most time there will be 3-7 regions, each one with 300-500 hundreds trees, will it make big difference by putting them in one container and use layer for these trees, and another layer for those player spine objs?

Thanks for feedback :)

**Comment by bigtimebuddy** [comment]
Transferring this to a discussion.
